returnn/tf/layers/rec.py

```diff
@@ -362,23 +362,22 @@ def get_out_data_from_opts(cls, network, unit, sources=(), initial_state=None, *
     """
     from tensorflow.python.util import nest
     source_data = get_concat_sources_data_template(sources) if sources else None
-    if source_data and not source_data.have_time_axis():
-      # We expect to be inside another RecLayer, and should do a single step (like RnnCellLayer).
-      out_time_dim_axis = None
-      out_batch_dim_axis = 0
-    else:
-      out_time_dim_axis = 0
-      out_batch_dim_axis = 1
+    has_time = False if source_data and not source_data.have_time_axis() else True
     n_out = kwargs.get("n_out", NotSpecified)
     out_type = kwargs.get("out_type", None)
     loss = kwargs.get("loss", None)
     deps = list(sources)  # type: typing.List[LayerBase]
     deps += [layer for layer in nest.flatten(initial_state) if isinstance(layer, LayerBase)]
     if out_type or n_out is not NotSpecified or loss:
       if out_type:
-        assert out_type.get("time_dim_axis", out_time_dim_axis) == out_time_dim_axis
-        assert out_type.get("batch_dim_axis", out_batch_dim_axis) == out_batch_dim_axis
+        assert out_type.get("time_dim_axis", 0 if has_time else None) == 0 if has_time else None
+        assert out_type.get("batch_dim_axis", 1 if has_time else 0) == 1 if has_time else 0
```

Comment:
I think it would be easier to just move that to the end, and then write
```suggestion
        assert out_type.get("time_dim_axis", out.time_dim_axis) == out.time_dim_axis
        assert out_type.get("batch_dim_axis", out.batch_dim_axis) == out.batch_dim_axis
```
That's much simpler and has less assumptions.

Reply:
Ah yes, that is definitely better